src/test/java/org/broadinstitute/hellbender/engine/dataflow/RefAPISourceUnitTest.java

```diff
@@ -0,0 +1,59 @@
+package org.broadinstitute.hellbender.engine.dataflow;
+
+import com.google.cloud.dataflow.sdk.Pipeline;
+import com.google.cloud.dataflow.sdk.options.PipelineOptionsFactory;
+import com.google.cloud.dataflow.sdk.testing.TestPipeline;
+import com.google.cloud.genomics.dataflow.utils.GenomicsOptions;
+import org.broadinstitute.hellbender.engine.dataflow.datasources.RefAPIMetadata;
+import org.broadinstitute.hellbender.engine.dataflow.datasources.RefAPISource;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.reference.ReferenceBases;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.Map;
+
+public class RefAPISourceUnitTest extends BaseTest {
+
+    private ReferenceBases queryReferenceAPI( final String referenceName, final SimpleInterval interval ) {
+        GenomicsOptions options = PipelineOptionsFactory.create().as(GenomicsOptions.class);
+        options.setApiKey(getDataflowTestApiKey());
+        options.setProject(getDataflowTestProject());
+
+        final Pipeline p = TestPipeline.create(options);
```

Comment:
(1) I think always prefer GATKTestPipeline() it doesn't matter if it takes command line args or not.
(2)  If it fails on spark it will fail on spark, but ideally hellbender should run unmodified on spark, so it's a bug if it can't.  It may make sense to have a different option for getting the reference on your local cluster, but it should still work with the google reference even if it's very slow.